packages/back-end/src/controllers/features.ts

```diff
@@ -756,17 +758,27 @@ export async function postFeaturePublish(
 
   // If changing the default value, it affects all enabled environments
   if (mergeResult.result.defaultValue !== undefined) {
-    req.checkPermissions(
-      "publishFeatures",
-      feature.project,
-      getEnabledEnvironments(feature, environmentIds)
-    );
+    if (
+      !context.permissions.canPublishFeature(
+        feature,
+        Array.from(getEnabledEnvironments(feature, environmentIds))
+      )
+    ) {
+      context.permissions.throwPermissionError();
+    }
   }
   // Otherwise, only the environments with rule changes are affected
   else {
     const changedEnvs = Object.keys(mergeResult.result.rules || {});
     if (changedEnvs.length > 0) {
-      req.checkPermissions("publishFeatures", feature.project, changedEnvs);
+      if (
+        !context.permissions.canPublishFeature(
+          feature,
+          Array.from(getEnabledEnvironments(feature, changedEnvs))
```

Comment:
Is this change intentional?  Went from just using `changedEnvs` to using `getEnabledEnvironments(feature, changedEnvs)`

Reply:
No - good catch.